packages/cli/src/decorators/on-shutdown.ts

```diff
@@ -33,8 +33,6 @@ export const OnShutdown =
 			Container.get(ShutdownService).register(priority, { serviceClass, methodName });
 		} else {
 			const name = `${serviceClass.name}.${methodName}()`;
-			throw new ApplicationError(
-				`${name} must be a method on ${serviceClass.name} to use "OnShutdown"`,
-			);
+			throw new UserError(`${name} must be a method on ${serviceClass.name} to use "OnShutdown"`);
```

Comment:
Do we need a `DeveloperError`?

Reply:
This could be a `RuntimeError`.

Reply:
`UnexpectedError` would be a `DeveloperError`